src/config/lockfile_entry_manager.py

```diff
@@ -0,0 +1,220 @@
+        """Returns a set containing the asset hashes that exist in the lockfile
+        but not in the current assets.
+
+        Args:
+            asset: The asset type to compare hashes for
+
+        Returns:
+            A set containing the set difference of the lockfile entries and the
+            current assets for the given asset type.
+        """
+        index = HASH_ALGS.index(DEFAULT_ALG)
+        lockfile_assets_set = set(self._get_nth_key_from_multikey(
+            self.lockfile_entries[asset], index))
+        curr_assets_set = set(self.curr_asset_hash_map[asset].keys())
+        return lockfile_assets_set.difference(curr_assets_set)
+
+    def _get_diff_of_curr_assets_and_lockfile(self, asset: str):
+        """Returns a set containing the asset hashes that exist in the current 
+        assets but not in the lockfile.
+
+        Args:
+            asset: The asset type to compare hashes for
+
+        Returns:
+            A set containing the set difference of the current assets and the 
+            lockfile entries for the given asset type.
+        """
+        index = HASH_ALGS.index(DEFAULT_ALG)
+        lockfile_assets_set = set(self._get_nth_key_from_multikey(
+            self.lockfile_entries[asset], index))
+        curr_assets_set = set(self.curr_asset_hash_map[asset].keys())
+        return curr_assets_set.difference(lockfile_assets_set)
+
+    def _get_intersect_of_lockfile_and_curr_assets(self, asset: str):
+        """Returns a set containing the asset hashes that exist in both the 
+        lockfile and the current assets.
+
+        Args:
+            asset: The asset type to compare hashes for
+
+        Returns:
+            A set containing the set intersect of the lockfile entries and the 
+            current assets for the given asset type.
+        """
+        index = HASH_ALGS.index(DEFAULT_ALG)
+        lockfile_assets_set = set(self._get_nth_key_from_multikey(
+            self.lockfile_entries[asset], index))
+        curr_assets_set = set(self.curr_asset_hash_map[asset].keys())
+        return lockfile_assets_set.intersection(curr_assets_set)
+
+    def apply_lockfile_for_asset(self, asset: str, install_queue: set):
+        """Given a set containing hashes of assets in the lockfile but not in 
+        the current assets, applies the state of the lockfile by downloading 
+        those assets.
+
+        Args:
+            asset: The asset type to apply the lockfile state for
+            install_queue: The set containing hashes of assets to download
+        """
+        for asset_sha in install_queue:
+            asset = self._get_asset_entry_from_hash(asset, asset_sha)
+            cdn_link = asset.file_data.cdn_link
+            asset_entry = self.entry_hash_map[asset].get(asset_sha)
+            asset_name = asset_entry.file_name
+            dest = Path(self.asset_paths[asset]) / asset_name
+            download_file(cdn_link, dest)
```

Comment:
Gather all of the cdn links, then pass it to a helper which will download them with subprocesses since these are network bottlenecked actions